collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -41,32 +42,43 @@
 @SuppressLint("ViewConstructor")
 public class UrlWidget extends QuestionWidget implements ButtonWidget {
 
-    private Uri uri;
-    final Button openUrlButton;
-    final TextView stringAnswer;
     private final CustomTabHelper customTabHelper;
+    private Uri uri;
 
-    public UrlWidget(Context context, QuestionDetails questionDetails) {
+    protected Button openUrlButton;
+    protected TextView stringAnswer;
+
+    public UrlWidget(Context context, QuestionDetails questionDetails, CustomTabHelper customTabHelper) {
         super(context, questionDetails);
+        this.customTabHelper = customTabHelper;
+    }
 
-        openUrlButton = createSimpleButton(getContext(), getFormEntryPrompt().isReadOnly(), context.getString(R.string.open_url), getAnswerFontSize(), this);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        ViewGroup answerView = (ViewGroup) LayoutInflater.from(context).inflate(R.layout.url_widget_answer, null);
 
-        stringAnswer = getCenteredAnswerTextView(getContext(), getAnswerFontSize());
+        openUrlButton = answerView.findViewById(R.id.url_button);
```

Comment:
Since we're going to be touching every widget, should we consider using view binding?

Reply:
Yes, actually. I didn't realize I can directly use them without view binding. And as a lot of tests have to be added, this is a better option.

Reply:
Just a small wording correction, you're actually using the [view binding feature](https://developer.android.com/topic/libraries/view-binding) now rather than explicitly binding your views. I know, it's confusing. Speaking of confusing, where is view binding turned on, @seadowg? I'm confused about not seeing it in the build.gradle file.

Reply:
I'm also confused. It looks like @SaumiaSinghal is just using `findViewById` with `protected` fields. @lognaturel just to be clear are you saying we should move to using the new `View Binding` feature before we accept this PR? I'd be in favour of that if so.

Reply:
Yes @seadowg :/ I was. I fixed that.

Reply:
Sorry, now I'm really confused, too! Yes, @seadowg, that's what I was requesting. I was expecting not to see any more explicit `findViewById` in the final code but they're still there.